composer.json

```diff
@@ -12,34 +12,34 @@
         }
     ],
     "require": {
-        "php": "~5.3|~7.0",
-        "symfony/framework-bundle": "~2.3|~3.0",
-        "symfony/routing": "~2.3|~3.0",
-        "symfony/http-kernel": "^2.3.24|~3.0",
-        "symfony/dependency-injection": "~2.3|~3.0",
-        "symfony/process": "~2.3|~3.0",
-        "symfony/finder": "~2.3|~3.0",
+        "php": "^7.1.3",
+        "symfony/framework-bundle": "~4.0",
+        "symfony/routing": "~4.0",
+        "symfony/http-kernel": "~4.0",
+        "symfony/dependency-injection": "~4.0",
+        "symfony/process": "~4.0",
+        "symfony/finder": "~4.0",
         "jms/aop-bundle": "~1.1",
-        "jms/metadata": "~1.0"
+        "jms/metadata": "~1.0",
+        "doctrine/annotations": "~1.0"
     },
     "require-dev": {
-        "symfony/validator": "~2.3|~3.0",
-        "symfony/form": "~2.3|~3.0",
-        "symfony/class-loader": "~2.3|~3.0",
-        "symfony/yaml": "~2.3|~3.0",
-        "symfony/browser-kit": "~2.3|~3.0",
-        "symfony/security-bundle": "~2.3|^3.0",
-        "symfony/expression-language": "~2.6|~3.0",
-        "symfony/twig-bundle": "~2.3|~3.0",
-        "sensio/framework-extra-bundle": "~2.0|~3.0",
-        "jms/security-extra-bundle": "~1.0",
+        "symfony/validator": "~4.0",
+        "symfony/form": "~4.0",
+        "symfony/class-loader": "~4.0",
+        "symfony/yaml": "~4.0",
```

Comment:
This is quite restrictive, we should keep supporting 3.x for now imo. Could you instead use 4.0 classes only if they're available?

Reply:
Maybe you can create a new branch for 4.x, and mark release version as 2.0.x.
Supporting 3.x and 4.x both needs some workarounds.